docs/Setting-up-monorepo.md

```diff
@@ -0,0 +1,357 @@
+
+By centralizing these utilities, you reduce code duplication and simplify maintenance, ensuring that any updates or bug fixes are immediately available to all your apps.
+
+In this setup, we’ll create a React Native app, a React app and a form-validator utility that includes functions to validate email addresses, passwords, and generic text fields. This utility will then be integrated to both apps.
+
+## Step 1: Setting up the monorepo
+
+First, follow the [Expo documentation on setting up monorepos](https://docs.expo.dev/guides/monorepos/) to initialize your own monorepo. This will include setting up your `packages/` and `apps/` directories and configuring Yarn workspaces.
+
+Here's a quick recap:
+
+1. Initialize the monorepo:
+
+```shell
+mkdir monorepo-example
+cd monorepo-example
+yarn init
+```
+
+2. Configure workspaces in `package.json`:
+
+```json
+{
+  "private": true,
+  "workspaces": {
+    "packages": [
+      "packages/*"
+    ],
+    "apps": [
+      "apps/*"
+    ]
+  }
+}
+```
+
+3. Create directory structure:
+
+```shell
+mkdir apps packages
+```
+
+## Step 2: Create mobile app using Ignite
+
+Ignite is a powerful tool for scaffolding React Native projects. In this step, we'll create your first app within the monorepo using Ignite.
+
+1. Install Ignite CLI (if you haven't already):
+
+```shell
+npx ignite-cli@latest
+```
+
+2. Generate a new app:
+   Navigate to the apps/ directory and run the following command to create a new app:
+
+```shell
+cd apps
+ignite new mobile
+```
+
+We suggest the following answers to the prompts:
+
+```
+📝 Do you want to use Expo?: Expo - Recommended for almost all apps [Default]
+📝 Which Expo workflow?: Expo Go - For simple apps that don't need custom native code [Default]
+📝 Do you want to initialize a git repository?: No
+📝 Remove demo code? We recommend leaving it in if it's your first time using Ignite: No
+📝 Which package manager do you want to use?: yarn
+📝 Do you want to install dependencies?: No
+```
+
+3. Open `metro.config.js` file
+
+```shell
+touch mobile/metro.config.js
+```
+
+4. Replace Metro configuration with the following content
+
+```
+const { getDefaultConfig } = require('expo/metro-config');
+const path = require('path');
+
+// Find the project and workspace directories
+const projectRoot = __dirname;
+// This can be replaced with `find-yarn-workspace-root`
+const monorepoRoot = path.resolve(projectRoot, '../..');
+
+const config = getDefaultConfig(projectRoot);
+
+// 1. Watch all files within the monorepo
+config.watchFolders = [monorepoRoot];
+// 2. Let Metro know where to resolve packages and in what order
+config.resolver.nodeModulesPaths = [
+  path.resolve(projectRoot, 'node_modules'),
+  path.resolve(monorepoRoot, 'node_modules'),
+];
+
+module.exports = config;
+```
+
+## Step 3: Create web app
+
+Let's create a second app now. This app is meant to be run on web browsers.
```

Comment:
It might be more useful to have something like a design system package in `/packages` that exports some components as the example, as that's far more common in my experience.